Web_Development/FrontEnd/CSS Selectors and Pseudo Selector/CSS_Selectors.md

```diff
@@ -0,0 +1,239 @@
+
+```CSS
+h1{
+    text-align: center;
+    color: green;
+}
+
+h2{
+    text-align: center;
+    color: green;
+}
+
+p{
+    text-align: center;
+    color: green;
+}
+```
+
+> IT can be perform in a short method using Group selectors
+
+```CSS
+p,h1,h2{
+    text-align: center;
+    color: green;
+}
+```
+
+---
+
+
+#### Pseudo Selectors:
+
+The **Pseudo Selectors** are used to target an elemnt when any action is performed:
+
+- checked
+- after
+- hover
+- visited
+- not(selector)
+
+```CSS
+.box:hover{
+    box-shadow: gray 2px 2px 20px;
+    position: relative;
+    font-size: 1.5rem;
+}
+
+.hide:after {
+    content: "";
+    display: block;
+    clear: both;
+    visibility: hidden;
+    font-size: 0;
+    height: 0;
+    }
+```
+
+These properties will work on element when we hover cuser on it.
+
+`not(selector) is used to remove a specific target from similer selected elements. Here selector is our id or class selectors usally.`
+
+<br>
+
+> #### Select some part of element using pseudo selectors:
+
+```CSS
+p::first-line {
+   font-weight: bold;
+   font-size: 1.2em;
+}
+
+p::first-letter {
+   float: left;
+   font-size: 2em;
+   font-weight: bold;
+   font-family: cursive;
+   padding-right: 2px;
+}
+
+```
+
+---
+<br>
+
+## Combinators:
+
+#### Child Combinators (>):
+The child combinator selects direct children of first element.
+###### CSS code
+```CSS
+div > p {
+    color: green;
+}
+```
+This property will apply on the paragraphs that are direct child of div.
+###### Html code
+```HTML
+<div>
+    <p id="direct-child">
+        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quae animi eos eum.
+    </p>
+</div>
+<p id = "para">
+    Error distinctio corrupti ipsum voluptatibus nostrum! Dicta nostrum culpa veritatis perferendis.
+</p>
+```
+Here above property will apply on the paragraph with id "direct-child" because it is a direct child of 'div', and not apply on the paragraph with id "para".
+
+#### Sibling Combinator (~):
+With this combinator we can select any sibling element; that means, it does not matter that elememt is direct child or not if its under selected tag it can be targeted. let's understand this with an example.
+
+###### Html code
+```HTML
+<div>
+    <section>
+        <p>
+            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id itaque dolores inventore?
+        </p>
+    </section>
+    <p>
+        nam soluta ipsa ex incidunt laudantium similique fugit perferendis sunt fuga cumque impedit tempore nobis porro a perspiciatis!
+    </p>
+</div>
+```
+###### CSS code
+```CSS
+div~p
+{
+    color: red;
+}
+```
+This property will apply on both paragraph unless one is not a direct-child.
+
+####  Adjacent Sibling Selector (+):
+This Select the element that is just after the another specific element.
+``` CSS
+div + p
+{
+    color: white;
+    background-color: orange;
+    font-style: bold;
+}
+```
+This will apply on the paragraphs that is just after the div.
+
+#### 
+
+
```

Comment:
Remove these lines.

Reply:
I have done advised changes please check it now.